sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java

```diff
@@ -143,10 +143,20 @@ public void recordGatewayResponse(
                 this.gatewayStatistics.requestCharge = storeResponse
                                                            .getHeaderValue(HttpConstants.HttpHeaders.REQUEST_CHARGE);
                 this.gatewayStatistics.requestTimeline = DirectBridgeInternal.getRequestTimeline(storeResponse);
+                backendLatencyInMsHeaderValue = storeResponse
+                    .getHeaderValue(HttpConstants.HttpHeaders.BACKEND_REQUEST_DURATION_MILLISECONDS);
+                if (StringUtils.isNotEmpty(backendLatencyInMsHeaderValue)) {
+                    this.gatewayStatistics.backendLatencyInMs = Double.parseDouble(backendLatencyInMsHeaderValue);
+                }
             } else if (exception != null) {
                 this.gatewayStatistics.statusCode = exception.getStatusCode();
                 this.gatewayStatistics.subStatusCode = exception.getSubStatusCode();
                 this.gatewayStatistics.requestTimeline = this.transportRequestTimeline;
+                backendLatencyInMsHeaderValue = exception.getResponseHeaders()
+                                                         .get(HttpConstants.HttpHeaders.BACKEND_REQUEST_DURATION_MILLISECONDS);
+                if (StringUtils.isNotEmpty(backendLatencyInMsHeaderValue)) {
+                    this.gatewayStatistics.backendLatencyInMs = Double.parseDouble(backendLatencyInMsHeaderValue);
+                }
```

Comment:
nit: could be extracted into a method as the same code is duplicated in two places?

Reply:
Yes could be done, but since it was just 2 lines of code, didn't want to make an extra function for it.